twisted/python/compat.py

```diff
@@ -765,6 +765,29 @@ def _coercedUnicode(s):
 
 
 
+def _bytesRepr(bytestring):
+    """
+    Provide a repr for a byte string that begins with 'b' on both
+    Python 2 and 3.
+
+    @param bytestring: The string to repr.
+    @type bytestring: L{bytes}
+
+    @raise TypeError: The input is not L{bytes}.
+
+    @return: The repr with a leading 'b'.
+    @rtype: L{bytes}
+    """
+    if not isinstance(bytestring, bytes):
+        raise TypeError("Expected bytes not %r" % (bytestring,))
```

Comment:
I guess that the intention was to get both the type and content from the repr() part.

for printing errors like this I usually truncate the input data at 50 chars ... so that if you use it for 16kb you will not get an excessive error message